proxy/http2/Http2ConnectionState.cc

```diff
@@ -1454,6 +1454,10 @@ Http2ConnectionState::send_push_promise_frame(Http2Stream *stream, URL &url)
   h1_hdr.create(HTTP_TYPE_REQUEST);
   h1_hdr.url_set(&url);
   h1_hdr.method_set("GET", 3);
+  if (accept_encoding != nullptr) {
+    h1_hdr.field_attach(accept_encoding);
```

Comment:
It seems like this line doesn't work because the MIMEField (`accept_encoding`) is attached to the original MIMEHdr (`mh` in `TSHttpTxnServerPush`). Probably we need to copy the field first.

Reply:
that makes sense. Can you point me to any example on how to copy fields? I don't have much experience manipulating headers yet and I'm not having luck figuring it out 😔

Reply:
Here is a place copying headers.
https://github.com/apache/trafficserver/blob/7.1.x/proxy/http2/HTTP2.cc#L575-L580